### PR TITLE
[ONNX] export remainder

### PR DESCRIPTION
**Pull request by neginraoof**
Added ONNX export support for torch.remainder and torch.fmod

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1900,3 +1900,12 @@ def multinomial(g, input, num_samples, replacement=False, generator=None):
     return g.op("Multinomial", log_input,
                 dtype_i=sym_help.cast_pytorch_to_onnx['Long'],
                 sample_size_i=num_samples)
+
+
+def remainder(g, input, other):
```
[the remark this answers is not in the document]
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Using Mod from onnx in fmod the sign of the remainder will be the same as the dividend. In PyTorch, however, the remainder has the same sign as the divisor.

[the remark this answers is not in the document]
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Sure.

[the remark this answers is not in the document]
Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py]: FYI There is `onnx::Mod` available https://github.com/onnx/onnx/blob/master/docs/Operators.md#mod, is it not suited for `torch.fmod`, `torch.remainder`?

**Review by BowenBao — CHANGES_REQUESTED** [review]
see inline comments

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: No. For onnx:Mod if fmod = 1, the sign of the remainder will be the same as the dividend. Which is not what we want in this case.

[the remark this answers is not in the document]
Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py]: I see, I also noticed that `onnx::Mod` was added in opset 10, so we need this op composition for opset 9 anyways.

There is a `fmod` attribute in `onnx::Mod`, however it is enforced to be 1 for floating point inputs, maybe we should relax this constraint in opset 11?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1900,3 +1900,12 @@ def multinomial(g, input, num_samples, replacement=False, generator=None):
...
+    div = g.op("Div", input, other)
+    if input.type().scalarType() != 'Int':
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py]: What about other integral types? (Long, uint8, etc)

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -1900,3 +1900,12 @@ def multinomial(g, input, num_samples, replacement=False, generator=None):
...
+        div = g.op("Floor", div)
+    quo = g.op("Mul", div, other)
+    dtype = input.type().scalarType()
+    return g.op("Cast", g.op("Sub", input, quo), to_i=sym_help.cast_pytorch_to_onnx[dtype])
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py]: Is this `Cast` needed? For `onnx::Sub`, it is enforced that inputs and output all share the same type, so the type of output from `g.op("Sub", input, quo)` must be the same as type of `input`, or else the ONNX graph is invalid.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -759,6 +759,24 @@ def forward(self, x):
...
+        x = torch.randn(4, 2, 3)
+        y = torch.randn(1, 2, 1)
+        self.run_test(RemainderModel(), (x, y))
+
+    def test_remainder_scalar(self):
+        class RemainderModel(torch.nn.Module):
+            def forward(self, input, other):
+                return torch.remainder(input, other)
+
+        x = torch.randint(10, (2, 3)).to(dtype=torch.int)
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: Try this example `torch.remainder(torch.tensor(10), 1.5)`

File: torch/onnx/symbolic_opset9.py (hunk 3 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: The purpose was to finally cast the output to the same type as input. I fixed this by initially casting 'other' to the same type as input.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 4 above)
Comment (neginraoof) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: This is supported now

Comment (neginraoof) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Added to the tests

**Review by BowenBao — COMMENTED** [review]
The remainder part looks good. Can we add fmod export as well? (export as onnx::mod for opset10 and after)

**Comment by neginraoof** [comment]
@BowenBao Could you please verify the changes requested?

**Comment by neginraoof** [comment]
@bddppq for review

**Review by BowenBao — APPROVED** [review]

**Comment by neginraoof** [comment]
cc @houseroad

**Comment by neginraoof** [comment]
cc @bddppq @houseroad for review

File: torch/onnx/symbolic_opset10.py [hunk 5]
```diff
@@ -196,3 +196,9 @@ def flip(g, input, dims):
                                   starts=[-1] * len(dims),
                                   ends=[-9223372036854775807] * len(dims),
                                   steps=[-1] * len(dims))
+
+
+def fmod(g, input, other):
+    dtype = input.type().scalarType()
+    other = g.op("Cast", sym_help._maybe_get_scalar(other), to_i=sym_help.cast_pytorch_to_onnx[dtype])
```
Comment (houseroad) [inline torch/onnx/symbolic_opset10.py]: Has @BowenBao 's scalar type analysis pass covered fmod op or not?

**Review by houseroad — APPROVED** [review]
LGTM

**Comment by neginraoof** [comment]
@houseroad I just fixed conflicts

**Comment by neginraoof** [comment]
cc @houseroad for review

**Review by houseroad — COMMENTED** [review]
Please update TestOperators.test_remainder's expect files.

**Comment by neginraoof** [comment]
cc @houseroad  for review

**Comment by neginraoof** [comment]
cc @houseroad for review

**Pull request closed without merging**